core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/ImmutablesTaster.java

```diff
@@ -0,0 +1,286 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper.reflect.internal;
+
+import java.lang.annotation.Annotation;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.jdbi.v3.core.config.JdbiConfig;
+import org.jdbi.v3.core.generic.GenericTypes;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoProperties.PojoProperty;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+
+public class ImmutablesTaster implements Function<Type, Optional<? extends PojoProperties<?>>>, JdbiConfig<ImmutablesTaster> {
+
+    private final Map<Class<?>, Function<Type, PojoProperties<?>>> registered = new HashMap<>();
+
+    public ImmutablesTaster() {}
+
+    private ImmutablesTaster(ImmutablesTaster other) {
+        registered.putAll(other.registered);
+    }
+
+    @Override
+    public ImmutablesTaster createCopy() {
+        return new ImmutablesTaster(this);
+    }
+
+    public <T> void register(Class<T> iface, Class<? extends T> impl, Class<? extends T> modifiable) {
+        final Function<Type, PojoProperties<?>> immutable = t -> new ImmutablePojoProperties<>(t, iface, impl);
+        registered.put(iface, immutable);
+        registered.put(impl, immutable);
+        if (modifiable != null) {
+            registered.put(modifiable, t -> new ModifiablePojoProperties<>(t, iface, modifiable));
+        }
+    }
+
+    @Override
+    public Optional<? extends PojoProperties<?>> apply(Type t) {
+        final Class<?> erased = GenericTypes.getErasedType(t);
+        return Stream.concat(Stream.of(erased), Arrays.stream(erased.getInterfaces()))
+                .map(registered::get)
+                .filter(Objects::nonNull)
+                .map(f -> f.apply(t))
+                .findAny();
+    }
+
+    static MethodHandle alwaysSet() {
+        return MethodHandles.dropArguments(MethodHandles.constant(boolean.class, true), 0, Object.class);
+    }
+
+    private static <T> T guard(ThrowingSupplier<T> supp) {
+        try {
+            return supp.get();
+        } catch (RuntimeException | Error e) { // NOPMD
+            throw e;
+        } catch (Throwable t) {
```

Comment:
The problem is that `MethodHandle.invoke` is declared as `throws Throwable`.
"Classic" reflection avoided this with `UndeclaredThrowableException` but that's gone from the "new" style.

Consider a property method
```java
public int getFoo() throws Throwable {
    throw new Throwable("barf");
}
```

this will be thrown as-is out of the `invoke` call and we have to handle that.  Or am I missing something obvious?

Reply:
This is all gone now in favor of `Unchecked`.